skimage/restoration/_denoise.py

```diff
@@ -594,26 +600,39 @@ def denoise_wavelet(image, sigma=None, wavelet='db1', mode='soft',
             ('Invalid method: {}. The currently supported methods are '
              '"BayesShrink" and "VisuShrink"').format(method))
 
-    image = img_as_float(image)
-
     if multichannel:
         if isinstance(sigma, numbers.Number) or sigma is None:
             sigma = [sigma] * image.shape[-1]
 
+    clip_output = False
+    if image.dtype.kind != 'f':
+        clip_output = True
+        data_range_pre = image.max() - image.min()
+        image = img_as_float(image)
+        data_range_post = image.max() - image.min()
+        scale_factor = data_range_post / data_range_pre
+        if sigma is not None:
+            if multichannel:
+                sigma = [s * scale_factor for s in sigma]
+            else:
+                sigma *= scale_factor
+
     if multichannel:
         if convert2ycbcr:
             out = color.rgb2ycbcr(image)
             for i in range(3):
                 # renormalizing this color channel to live in [0, 1]
                 min, max = out[..., i].min(), out[..., i].max()
                 channel = out[..., i] - min
-                channel /= max - min
+                sf = max - min
```

Comment:
I'd rename this to `scale_factor`, and also, I know this isn't this PR's issue, but we should aim not to override Python built-in names, so `min` and `max` should probably be renamed to `channel_min` and `channel_max`, for example.

Reply:
Also, we should probably check for 0 range and exit early, rather than dividing by 0.

Reply:
Good idea on the range check. I think we can just skip "empty" channels via.
```python
                if scale_factor == 0:
                    # nothing to denoise if channel has zero amplitude
                    continue
```

Reply:
Actually, I need to revisit the whole YCbCr case and add tests for it. I think to do that color conversion we will need to rescale the floating-point RGB values to [0, 1] range first.
I am actually not sure why this per-channel rescaling after YCbCr conversion was used in the first place, but the author explicitly mentions in the docstring:

`
    When YCbCr conversion is done, every color channel is scaled between 0
    and 1, and `sigma` values are applied to these scaled color channels.
`

Reply:
Looking into `rgb2ycbcr`, it calls `_prepare_color_array` internally which uses `img_as_float` and thus does not automatically rescale floating point inputs. The docstring states that: `Y is between 16 and 235.`, but this is not true for floating point inputs unless they are in [0, 1].

example:
```Python
# uint8 case
np.max(skimage.color.rgb2ycbcr(data.chelsea()))
Out[75]: 182.74402352941178

# floats in range [0, 255.]
np.max(skimage.color.rgb2ycbcr(data.chelsea().astype(float)))
Out[76]: 42535.726
```

For that reason, it is good that we are also rescaling the YCbCr channels again in the inner channel denoising loop. The question is how to best handle sigma in this case:

Perhaps we should apply a global sigma scaling across all channels as opposed to doing per-channel sigma scaling as is being done currently in this PR.  That would be more consistent with the current YCbCr description in the denoise_wavelet docstring.

Reply:
I have updated the YCbCr case with a proper solution (see 7e25ef3) for scaling `sigma` using the color conversion matrix (assuming that the noise across the R, G, B channels is independently distributed).

Reply:
I made the changes suggested above by @jni here